src/olympia/translations/tests/test_commands.py

```diff
@@ -0,0 +1,70 @@
+from django.core.management import call_command
+
+from olympia.amo.tests import addon_factory, TestCase
+from olympia.bandwagon.models import Collection
+from olympia.translations.models import Translation
+from olympia.translations.management.commands.process_translations import (
+    Command as ProcessTranslationsCommand,
+)
+
+
+class TestTranslationCommands(TestCase):
+    def setUp(self):
+        self.collection = Collection.objects.create(
+            name='foo_collection_name', description='foo_collection_description'
+        )
+        # Orphaned translation: should not be touched.
+        Translation.objects.create(
+            id=667, localized_string='foo', localized_string_clean='bar', locale='de'
+        )
+        # Orphaned translation for the same string in a different locale:
+        # should not be touched.
+        Translation.objects.create(
+            id=667,
+            localized_string='foo2',
+            localized_string_clean='bar2',
+            locale='fr',
+        )
+
+        # Translation belonging to a collection name: should not be touched.
+        self.collection.name.localized_string_clean = 'bar_collection_name'
+        self.collection.name.save()
+        # Translation belonging to an add-on: should not be touched.
+        addon = addon_factory(name='foo_addon')
+        addon.name.localized_string_clean = 'bar_addon'
+        addon.name.save()
+        # Translation belonging to a collection description that we don't need to bother
+        # with (because its localized string clean should already match its localized
+        # string): should not be touched.
+        extra_collection = Collection.objects.create(
+            name='foo_collection_name2', description='foo_collection_description2'
+        )
+        extra_collection.description.update(modified=self.days_ago(42))
+
+        # The command should fix this one.
```

Comment:
Just for ease of readability of the test, I would suggest moving this block up right below where you create `self.collection`. Otherwise one has to scroll up to verify that `localized_string_clean` is in fact different from `description` in `self.collection`.

Reply:
7afa8a5dbaf2b10dd69842e110a44c2b35fe11e7